node-js-web-server/server230.js

```diff
@@ -0,0 +1,40 @@
+const http = require('http');
+const path = require('path');
+const fs = require('fs');
+
+http.createServer(function (request, response) {
+   console.log('request url: ', request.url);
+
+   var filePath = './public' + request.url;
+   if (filePath == './public/') {
+      filePath = './public/index.html';
+   }
+   console.log(filePath)
+
+   var extName = String(path.extname(filePath)).toLowerCase();
+   var mimeTypes = {
+      '.html': 'text/html',
+      '.js': 'text/javascript',
+      '.css': 'text/css',
+      '.png': 'image/png'
+   };
+   var contentType = mimeTypes[extName] || 'application/octet-stream';
+
+   fs.readFile(filePath, function(error, content) {
+      if (error) {
+         fs.readFile('./public/404.html', function(error, content) {
+            response.writeHead();
+        }); 
+      } else {
+         
+      }
+   });
+}).listen(8080);
+console.log('Server is running at http://localhost:8080');
+
+// `writeHead` method can be passed two arguments:
```

Comment:
The `writeHead` method takes two arguments: the status code of the response, and a headers object.
Pass `writeHead` 404 as its first argument, and `{'Content-Type': // file content-type}` as the second argument.
Remember that you can get the content type for the second argument from the `contentType` variable.